akka-docs/rst/scala/code/docs/http/scaladsl/WebSocketClientExampleSpec.scala

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2009-2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.scaladsl
+
+import akka.actor.ActorSystem
+import org.scalatest.{ Matchers, WordSpec }
+
+class WebSocketClientExampleSpec extends WordSpec with Matchers {
+
+  "single-WebSocket-request-example" in {
+    pending // compile-time only test
+    //#single-websocket-request
+    import akka.Done
+    import akka.http.scaladsl.Http
+    import akka.stream.ActorMaterializer
+    import akka.stream.scaladsl._
+    import akka.http.scaladsl.model._
+    import akka.http.scaladsl.model.ws._
+
+    import scala.concurrent.Future
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+    import system.dispatcher
+
+    // future done is the materialized value of Sink.foreach,
+    // emitted when the stream completes
+    val flow: Flow[Message, Message, Future[Done]] = Flow.fromSinkAndSourceMat(
+      Sink.foreach[Message] {
+        case message: TextMessage.Strict =>
+          println(message.text)
+      },
+      // send this as a message over the WebSocket
+      Source.single(TextMessage("hello world!")))(Keep.left)
+
+    // second value NotUsed in this case
+    val (upgradeResponse: Future[WebsocketUpgradeResponse], closed: Future[Done]) =
```

Comment:
maybe drop off the types here, its a bit harder to read IMO... doc types in text instead?